### ISSUE TITLE
alternative answers

### ISSUE THREAD

**Issue by Maximilianos** [rmurphey/js-assessment-answers#37]
What do you think of having other flavors of the answers? 
Clearly labelled of course.

for example: 
- answers using modern es6+ JavaScript syntax, 
- answers using current es5 syntax `.forEach` `.map` etc
- answers that are really really optimized for performance

PS. Really enjoyed the challenge of the test, thanks for putting it together!

**Comment by jmeas** [comment]
Thanks for the suggestions, @Maximilianos! These are definitely important skills, and I'd love to see more assessment of them in the tests.

I'm curious to hear what @ashleygwilliams / @kadamwhite think of including multiple versions of answers?

My immediate thought is that I really value the simplicity of the assessment right now. Keeping the questions and answers 1:1 makes it a lot easier for folks to go in and compare what they did with the answers we provide.

I'm wondering if devising specific tests to evaluate those skills – and clearly saying it in the question – would be a better way to clearly and better assess developers' knowledge of these specific areas.

But I'd also be fine just including multiple answers if everyone else thought that was a good idea. I'm curious to see how we'd go about implementing 'em.

Fwiw, I thought I'd mention that some of the tests do [have comments that show how to do it using ES5 methods](https://github.com/rmurphey/js-assessment-answers/blob/master/app/arrays.js#L5-L7).

**Comment by Maximilianos** [comment]
Yeah, I was between multiple branches of this repo vs multiple directories for the answers kinda like:
```
- app
  ┖- base // the current answers
  ┖- es6
  ┖- perf
```

I was just interested in seeing the "advanced" answers which would use all the tools currently in the arsenal of a JS developer. I do appreciate, of course, the instructive nature of the simple answers but I would have liked to have seen the other flavors of those same answers as well :)

As for your suggestion, in my opinion you shouldn't need to add specific tests to showcase es6 et al. as all the current tests can be answered using any syntax of JS you like.
